### ISSUE TITLE
Ability to report comments from within the messaging addon

### ISSUE THREAD

**Issue by BoomerScratch** [ScratchAddons/ScratchAddons#477]
**Is your feature request related to a problem? Please describe.**

Yesterday, someone on my profile replied to someone else's comment asking for personal information. I tried reporting the comment from within the messaging addon but I noticed I couldn't.

**Describe the solution you'd like**

Add a report button above comments when hovering over them.

**Comment by pufferfish101007** [comment]
I think this could be extremely useful! Support.

**Comment by WorldLanguages** [comment]
I agree, however I can't come up with a good way to test if a "report" implementation works, without sending dummy reports to the ST.

**Comment by pufferfish101007** [comment]
Actually, what we could do is open it up in a new tab, to remain consustent with Scratch, retain trust and no testing needed. It could be something like 
```
function reportComment (commentId, commentUrl) {
var reportWindow = window.open(commentUrl);
reportWindow.document.querySelector('selector of the comment report button').click();

}

**Comment by WorldLanguages** [comment]
@pufferfish101007 Hmm that could work, however it will only save the user a single click. They can already get a direct link to the comment, wait for it to load, then report it.

**Comment by DNin01** [comment]
Yeah, if you can delete messages from an addon, you should also be able to report them.

The option to report something is arguably more valuable. In fact, I don't remember ever having to delete a comment without reporting it. But with only a Delete button, users might resort to, well, deleting inappropriate comments instead of reporting them.

**Comment by WorldLanguages** [comment]
Yeah, let's add a Report link like pufferfish suggested.

**Comment by DNin01** [comment]
Is there already a way to open comments in a new tab on the Scratch website? If not, we could replace the Delete button with such a link, where you would have the option to report the comment.

**Comment by DNin01** [comment]
Just used it, turns out there's actually an "external link" icon next to the timestamp to open the comment in a new tab, so you could get to the comment there.

Since reporting is so much more useful than deleting, I'd suggest simply dropping the delete feature from the popup and instead users can open the comments in a new tab using that button to access them on the website and report them. Maybe we could make it more obvious that you can do that by replacing the "Delete" button with an "Open" button.

**Comment by WorldLanguages** [comment]
> Since reporting is so much more useful than deleting

Well, that is subjective :P

> I'd suggest simply dropping the delete feature from the popup

Hmm... Someone should check if the "delete" button was inherited from griffpatch's "Scratch Messaging Extension". If that's the case then we would be removing a 7 year old feature 🤔

**Comment by WorldLanguages** [comment]
Like I said 2 months ago, I believe we should implement pufferfish's suggestion. We can add our own Report button which opens the comment in a new tab, and clicks the Report button on behalf of the user. Then the rest of the reporting flow is in charge of Scratch itself.
Worst that can happen is that we fail to click the Report button on behalf of the user and they need to manually click it. But we already know how to open a comment in a new tab, as DNin said.

**Comment by WorldLanguages** [comment]
I'm adding a priority label:

**Comment by CST1229** [comment]
> I agree, however I can't come up with a good way to test if a "report" implementation works, without sending dummy reports to the ST.

Maybe find a real rulebreaking comment somehow and report it?
